src/components/DatePicker/DatePicker.vue

```diff
@@ -0,0 +1,305 @@
+<template>
+  <div
+    class="calendar"
+    @keyup.right="() => incrementMonth(true)"
+    @keyup.left="() => decrementMonth(true)"
+    tabindex="0"
+    ref="calendar"
+  >
+    <div class="calendar-header">
+      <button
+        class="month-button cursor-pointer"
+        aria-label="Previous month"
+        title="Previous month"
+        @click="() => decrementMonth()"
+      >
+        <i class="pi pi-arrow-left-s-line" />
+      </button>
+
+      <button
+        class="month-button cursor-pointer"
+        aria-label="Next month"
+        title="Next month"
+        @click="() => incrementMonth()"
+      >
+        <i class="pi pi-arrow-right-s-line" />
+      </button>
+
+      <div class="month-title ml-2"> {{ displayMonth }} {{ year }} </div>
+
+      <a
+        v-if="showToday"
+        class="today-link"
+        role="button"
+        tabindex="0"
+        @click="resetDate"
+        @keyup.enter.space="resetDate"
+      >
+        Today
+      </a>
+    </div>
+
+    <transition :name="monthDirection == 1 ? 'slide' : 'slide-reverse'">
+      <div class="days-grid" :key="month">
+        <div class="days-header-grid">
+          <div
+            v-for="(n, i) in 7"
+            class="day-header"
+            :class="'day-' + getDayOfTheWeek(i)"
+            :key="i"
+          >
+            {{ getDisplayDay(i) }}
+          </div>
+        </div>
+
+        <button
+          v-for="day in Array.from({ length: getDayOfTheWeek(1) })
+            .map((e, i) => daysInPreviousMonth - i)
+            .reverse()"
```

Comment:
Personally I'd like to see this logic abstracted into a well named getter. Something like `daysFromPreviousMonth`.